src/sage/meson.build

```diff
@@ -12,7 +12,6 @@ datadir = fs.expanduser(get_option('datadir'))
 if not fs.is_absolute(datadir)
   datadir = prefix / datadir
 endif
-conf_data.set('SAGE_SHARE', datadir)
```

Comment:
The idea of this construction was to allow distros to specify a custom search path for db via `-Ddatadir` (via https://github.com/sagemath/sage/pull/40894/files#diff-45c63bcc74fb1cd2c0781892587774cffa3bd405d659ec188dee4afe0d583f59R544). Why do you think this is not necessary/helpful anymore?